lib/net.js

```diff
@@ -586,7 +574,7 @@ Socket.prototype._read = function(n) {
 
 
 Socket.prototype.end = function(data, encoding, callback) {
-  ReflectApply(stream.Duplex.prototype.end, this, [data, encoding, callback]);
+  stream.Duplex.prototype.end.call(this, data, encoding, callback);
```

Comment:
@aduh95 note the difference here. `ReflectApply` was allocating an additional array.

Reply:
Yes, I agree using `ReflectApply` here was a mistake. Although, in most cases I've encountered, replacing it with `FunctionPrototypeCall` was good enough. I'll update the other PR.